File: src/main/java/org/activecraft/activecraftcore/manager/LockdownManager.java
package org.activecraft.activecraftcore.manager;

import org.activecraft.activecraftcore.ActiveCraftCore;
import org.activecraft.activecraftcore.events.LockdownEvent;
import org.activecraft.activecraftcore.playermanagement.Profilev2;
import org.activecraft.activecraftcore.ActiveCraftCore;
import org.activecraft.activecraftcore.playermanagement.Profilev2;
import org.bukkit.Bukkit;

public class LockdownManager {

    public static void lockdown(boolean lockdown) {
        //call event
        LockdownEvent event = new LockdownEvent(lockdown);
        Bukkit.getPluginManager().callEvent(event);
        if (event.isCancelled()) return;

        ActiveCraftCore.getInstance().getMainConfig().set("lockdown.enabled", lockdown);
        if (!lockdown) return;
        Bukkit.getOnlinePlayers().stream()
                .map(Profilev2::of)
                .filter(profile -> profile != null && !profile.canBypassLockdown())
                .forEach(profile -> profile.getPlayer().kickPlayer(
                        profile.getMessageSupplier(ActiveCraftCore.getInstance()).getMessage("command.lockdown.lockdown-message"))
                );
    }

}

File: src/main/java/org/activecraft/activecraftcore/manager/LockdownManager.kt
package org.activecraft.activecraftcore.manager

import org.activecraft.activecraftcore.ActiveCraftCore
import org.activecraft.activecraftcore.events.LockdownEvent
import org.activecraft.activecraftcore.playermanagement.Profile
import org.bukkit.Bukkit

object LockdownManager {
    fun lockdown(lockdown: Boolean) {
        val event = LockdownEvent(lockdown)
        Bukkit.getPluginManager().callEvent(event)
        if (event.cancelled) return
        ActiveCraftCore.INSTANCE.mainConfig["lockdown.enabled"] = lockdown
        if (!lockdown) return
        Bukkit.getOnlinePlayers()
            .map { Profile.of(it) }
            .filter { !it.bypassLockdown }
            .forEach {
                it.player!!.kickPlayer(
                    it.getMessageSupplier(ActiveCraftCore.INSTANCE)!!.getMessage("command.lockdown.lockdown-message")
                )
            }
    }
}